src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -34,64 +34,78 @@
 import ai.rapids.cudf.Table;
 
 public class GpuTimeZoneDB {
-
-  public static final int TIMEOUT_SECS = 300;
-
+  private boolean cacheEmpty = true;
 
   // For the timezone database, we store the transitions in a ColumnVector that is a list of 
   // structs. The type of this column vector is:
   //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
-  private CompletableFuture<Map<String, Integer>> zoneIdToTableFuture;
-  private CompletableFuture<HostColumnVector> fixedTransitionsFuture;
+  private Map<String, Integer> zoneIdToTable;
+  private HostColumnVector fixedTransitions;
 
-  private boolean closed = false;
-
-  GpuTimeZoneDB() {
-    zoneIdToTableFuture = new CompletableFuture<>();
-    fixedTransitionsFuture = new CompletableFuture<>();
+  private GpuTimeZoneDB() {
   }
   
-  private static GpuTimeZoneDB instance = new GpuTimeZoneDB();
+  private static final GpuTimeZoneDB instance = new GpuTimeZoneDB();
   // This method is default visibility for testing purposes only. The instance will be never be exposed publicly
   // for this class.
   static GpuTimeZoneDB getInstance() {
     return instance;
   }
-  
+
+  /**
+   * This should be called on startup of an executor.
+   */
+  public static void cacheDatabaseAsync() {
+    Runnable runnable = () -> {
+      try {
+        cacheDatabase();
+      } catch (Exception e) {
+        // ignore
```

Comment:
Do we really want to ignore any exceptions thrown here?  Why not let Spark's uncaught exception handler at least log it, or log it here?  Silently eating exceptions is very frustrating to debug.

Reply:
Done. Now log the error.